base/client.jl

```diff
@@ -292,6 +292,30 @@ function exec_options(opts)
         elseif cmd == 'E'
             invokelatest(show, Core.eval(Main, parse_input_line(arg)))
             println()
+        elseif cmd == 'm'
+            mods = split(arg, ".")
+            pkg = popfirst!(mods)
+            # Some more validation of `mods` and `pkg`?
+            m = Base.require(Main, Symbol(pkg))
```

Comment:
So, technically this is all that is required if the module exports its main function - that was one of the explicit design goals of `@main` that you could use it as `-e 'using Foo'` to get the python app like behavior. I think it's fine to further shortcut that to `-m`, but I'm not sure about duplicating the `main` invocation logic here.

Reply:
Okay I agree with that. So if an app want to have multiple "functionalities" (like the proposed `julia -m Foo/Bar` to run `Foo.Bar.main`) would you do that as `julia -m Foo --bar` or still do `julia -m Foo/Bar` and then have that still call `Foo.main` and look at something like `Base.PROGRAM_FILE` to determine what to execute.